BrainPortal/app/views/data_providers/_show_user_key.html.erb

```diff
@@ -0,0 +1,50 @@
+
+<%-
+#
+# NeuroHub Project
+#
+# Copyright (C) 2020
+# The Royal Institution for the Advancement of Learning
+# McGill University
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+-%>
+
+
+<div class="display_cell" style="padding-left: 1em;">
+    <table>
+      <tr><th class="supertitle"> Configuration instructions </th></tr>
+      <tr><td>
+        Download
+        <%= link_to "mykey.pub", "data:plain;charset=utf-8,#{url_encode(current_user.ssh_key(create_it: true).public_key) rescue "Error" }" , download: "mykey.pub" %>
+        file and open command line in the folder you downloaded the file (typically called Downloads).
+        <br>Then use the following command to add it to the authorized_keys file on the host. Remember to replace
+        REMOTE_USERNAME <br> and REMOTE_HOSTNAME stabs with the values that you provided in SSH parameters section
+      </td></tr>
+      <tr><th class="supertitle"> Command line </th></tr>
+      <tr><td>
+        <pre class="ssh_key"> cat mykey.pub | ssh REMOTE_USERNAME@REMOTE_HOSTNAME "mkdir -p ~/.ssh && chmod 700 ~/.ssh && cat >>  ~/.ssh/authorized_keys</pre>
+      </td></tr>
+      <tr><td>
+        If download did not work paste the ssh key bellow in a text file and rerun the command line
+      </tr></td>
+      <tr><th class="supertitle">Public SSH Key</th></tr>
+      <tr><td><pre class="ssh_key"><%= pretty_ssh_key @data_provider.user.ssh_key(create_it: true).public_key rescue "Error fetching public key" %></pre></td></tr>
```

Comment:
This always crashes and never show the key, because `@data_provider` is not defined in a partial.

Again: `Did you even try your own code?!?`